src/com/gh4a/activities/Github4AndroidActivity.java

```diff
@@ -103,9 +111,11 @@ protected void onNewIntent(Intent intent) {
                     .appendQueryParameter(PARAM_CODE, data.getQueryParameter(PARAM_CODE))
                     .build();
             new FetchTokenTask(uri).schedule();
-        } else {
-            super.onNewIntent(intent);
+            return true;
         }
+
+        super.onNewIntent(intent);
```

Comment:
This line should be removed, right? Otherwise the super method is called twice AFAICT.